packages/web3torrent/src/library/web3torrent-lib.ts

```diff
@@ -158,29 +183,67 @@ export default class WebTorrentPaidStreamingClient extends WebTorrent {
+
+      if (!this.amSeeder) {
+        // leecher is participants[0], seeder is participants[1]
+        this.channelClient.createChannel(
+          peerAccount, // seeder
+          pseAccount, // leecher
+          '0', // seederBalance
+          '50', // leecherBalance,
+          await this.channelClient.getEthereumSelectedAddress(), // seederOutcomeAddress,
+          '0x0' // leecherOutcomeAddress
+        );
+      }
+    });
+
     wire.paidStreamingExtension.on(PaidStreamingExtensionEvents.NOTICE, notice =>
       torrent.emit(PaidStreamingExtensionEvents.NOTICE, wire, notice)
     );
+
+    // If the wallet queues a message, send it across the wire
+    this.channelClient.onMessageQueued(({sender, recipient, data}) => {
+      wire.paidStreamingExtension.payment(JSON.stringify(data)); // TODO don't use 'payment' since the messages are mor general than that
+    });
   }
 
   protected loadFunds(infoHash: string, peerId: string, paymentHash: string) {
     // [ George ] Here the seeder can countersign the state update by an appropriate channelClient.updateChannel(), and pull the updated state channel balance off the ChannelResult before updating this.allowedPeers[infoHash][peerId].funds accordingly.
 
     // [ George ] If web3torrent is to run the Single Asset Payments ForceMoveApp, and the payments are going to be unidirectional we could wrap the channelClient in a web3tChannelClient which offers a countersign() convenience method. This allows the seeder to immediately accept the payment and for the leecher to be ready to send another one as quickly as possible.
 
-    // [ George ] NB all channel client methods are async so we would want to await them before continuing. That means a bunch of methods in this class will also need to be aysnc.
+    // [ George ] NB all channel channelClient methods are async so we would want to await them before continuing. That means a bunch of methods in this class will also need to be aysnc.
 
     const {funds} = this.allowedPeers[infoHash][peerId];
     this.allowedPeers[infoHash][peerId].funds = (Number(funds) + Number(paymentHash)).toString();
   }
 
-  protected transferFunds(wire: PaidStreamingWire) {
+  protected async transferFunds(wire: PaidStreamingWire) {
     // [ George ] I think this is where the leecher could call channelClient.updateChannel(). A first iteration might just do this without any UI or checks.
 
-    // INFO: Assumed payment. This could emit an event on the UI to ask for more funds, or be automatic. Dunno.
-    setTimeout(() => {
-      wire.paidStreamingExtension.payment('50');
-    }, 500);
+    const channel = await this.channelClient.createChannel(
+      '0x0000000000000000000000000000000000000000',
+      '0x0000000000000000000000000000000000000000',
+      '0x00',
+      '0x00',
+      '0x0000000000000000000000000000000000000000',
+      '0x0000000000000000000000000000000000000000'
+    );
+
```

Comment:
This doesn't really belong here, but we need the `channel` (in particular `channel.channelId` for the `updateChannel` call below (which I think *does* belong here).